diff --git a/ngapp/app/controller.js b/ngapp/app/controller.js
index daae5e3..c4db961 100644
--- a/ngapp/app/controller.js
+++ b/ngapp/app/controller.js
@@ -4,9 +4,9 @@
             max: 100
     })
     .controller("ApplicationController", ["$scope", "$http", "$location", "$timeout", "$rootScope","$localStorage","$sessionStorage", ApplicationController])
-    .controller("HomeController", ["$scope", "$route","$http","$localStorage", HomeController])
+    .controller("HomeController", ["$scope", "$route","$http","$routeParams", HomeController])
     .controller("ConfigController", ["$scope", "$route","$http","$localStorage", ConfigController])    
-    .controller("AdminController", ["$scope", "$route","$http","$timeout", AdminController])
+    .controller("AdminController", ["$scope", "$route","$http","$timeout", "$location", AdminController])
     .controller("SceneController", ["$scope", "$route","$http","$localStorage", SceneController])
     .controller("LoginController", ["$scope", "$route","$http","$localStorage", LoginController])
     .controller("UISelectController", ["$scope", "$attrs", "progressConfig", UISelectController])
diff --git a/ngapp/app/controller/ApplicationController.js b/ngapp/app/controller/ApplicationController.js
index 09efedd..b25be31 100644
--- a/ngapp/app/controller/ApplicationController.js
+++ b/ngapp/app/controller/ApplicationController.js
@@ -9,18 +9,9 @@
 	$scope.callBackpageLoading();
 */
 	$scope.isActive=function(pathMenu){
-            pathMenu = "/"+pathMenu;
-            if($location.path() == pathMenu){
-                    return true;
-            }
-            else{
-                    return false;
-            }
+            return ($location.path() == "/" + pathMenu) ? true : false;
 	}
-        $scope.environmentList = ["DEV", "QA", "ALT-PROD", "PROD"];
-	$scope.projectList = ["PROJ 1", "PROJ 2", "PROJ 3", "PROJ 4"];
-	$scope.portalList = ["PORTAL 1", "PORTAL 2"];
-
+        
         $rootScope.data = {
             userName: 'Yogesh Surendran',
             greetingKey: 'Mr. ',
@@ -81,32 +72,54 @@
             portalList: [],
             environmentList: [],
             projectList: [],
+            getEnvironmentId: function(value){
+                return $scope.application.getId($scope.application.environmentList, value, "environmentId", "environmentNm")
+            },
+            getEnvironmentId: function(value){
+                return $scope.application.getId($scope.application.environmentList, value, "environmentId", "environmentNm")
+            },
+            getProjectId: function(value){
+                return $scope.application.getId($scope.application.projectList, value, "moduleId", "moduleNm")
+            },
+            getPortalId: function(value){
+                return $scope.application.getId($scope.application.portalList, value, "portalId", "portalNm")
+            },
+            getId: function(list, value, idField, valueField){
+                var id = -1;
+                for(var i=0; i < list.length; i++){
+                    var obj = list[i];
+                    if(obj[valueField] === value){
+                        id = obj[idField];
+                    }
+                }
+                return id;
+            },
 	    setProject: false,
 	    removeProject: false,
 	    getProject: false,
 	    getEnvironment:false,
-		preventToolbar: false,
-		preventTabbar: false,
-		errorMsg:"",
-		futureMsg:"Thanks for your interest This option will be added as part of next release!",
-		notificationsCount:"",
-		model:{
-			errMsg:"",
-			visible: false,
-			toggle: function(temp){
-				$rootScope.application.model.errMsg = temp;
-				$rootScope.application.model.visible = !$rootScope.application.model.visible;
-			}
-		},
-		dosignout:{
-			sout: false,
-			toggle1: function(){
-				$rootScope.application.dosignout.sout = !$rootScope.application.dosignout.sout;
-				$localStorage.$reset();
-				localStorage.clear();
-			}
-		},
-		showSubPageToolbarBtn: false,
+            preventToolbar: false,
+            preventTabbar: false,
+            errorMsg:"",
+            futureMsg:"Thanks for your interest This option will be added as part of next release!",
+            notificationsCount:"",
+            model:{
+                    errMsg:"",
+                    visible: false,
+                    toggle: function(temp){
+                            $rootScope.application.model.errMsg = temp;
+                            $rootScope.application.model.visible = !$rootScope.application.model.visible;
+                    }
+            },
+            dosignout:{
+                    sout: false,
+                    toggle1: function(){
+                            $rootScope.application.dosignout.sout = !$rootScope.application.dosignout.sout;
+                            $localStorage.$reset();
+                            localStorage.clear();
+                    }
+            },
+            showSubPageToolbarBtn: false,
 	    slider:{
 	    	animated:false,
 	    	makeSlide: function(){
diff --git a/ngapp/app/router.js b/ngapp/app/router.js
index b500f34..ecb984e 100644
--- a/ngapp/app/router.js
+++ b/ngapp/app/router.js
@@ -3,22 +3,14 @@
 
 function routerFn($routeProvider, $locationProvider) {
     $routeProvider
-            .when('/Home', {templateUrl: 'app/view/Home.html', controller: 'HomeController'})
-            .when('/Home/:subpage', {templateUrl: 'app/view/Home.html', controller: 'HomeController'})
-            .when('/Login', {templateUrl: 'app/view/Login.html', controller: 'LoginController'})
-            .when('/Admin', {templateUrl: 'app/view/Admin.html', controller: 'AdminController'})
-            .otherwise({
-                    redirectTo:'/Home/select'
-            });
+        .when('/Home', {templateUrl: 'app/view/Home.html', controller: 'HomeController'})
+        .when('/Home/:subpage', {templateUrl: 'app/view/Home.html', controller: 'HomeController'})
+        .when('/Chart/:id/:name', {templateUrl: 'app/view/Home.html', controller: 'HomeController'})
+        .when('/Login', {templateUrl: 'app/view/Login.html', controller: 'LoginController'})
+        .when('/Admin', {templateUrl: 'app/view/Admin.html', controller: 'AdminController'})
+        .otherwise({
+                redirectTo:'/Admin'
+        });
     // configure html5 to get links working on jsfiddle
     $locationProvider.hashPrefix('!').html5Mode(false);
-}
-
-/*resolve: {
-    // I will cause a 1 second delay
-    delay: function($q, $timeout) {
-            var delay = $q.defer();
-            $timeout(delay.resolve, 1000);
-            return delay.promise;
-    }
-}*/
+}
\ No newline at end of file
