diff --git a/ngapp/app/controller.js b/ngapp/app/controller.js
index c4db961..b39f9fc 100644
--- a/ngapp/app/controller.js
+++ b/ngapp/app/controller.js
@@ -5,8 +5,8 @@
     })
     .controller("ApplicationController", ["$scope", "$http", "$location", "$timeout", "$rootScope","$localStorage","$sessionStorage", ApplicationController])
     .controller("HomeController", ["$scope", "$route","$http","$routeParams", HomeController])
-    .controller("ConfigController", ["$scope", "$route","$http","$localStorage", ConfigController])    
-    .controller("AdminController", ["$scope", "$route","$http","$timeout", "$location", AdminController])
+    .controller("ConfigController", ["$scope", "$route","$http","$localStorage", "$routeParams", ConfigController])    
+    .controller("AdminController", ["$scope", "$route","$http","$timeout", "$location", "$routeParams", AdminController])
     .controller("SceneController", ["$scope", "$route","$http","$localStorage", SceneController])
     .controller("LoginController", ["$scope", "$route","$http","$localStorage", LoginController])
     .controller("UISelectController", ["$scope", "$attrs", "progressConfig", UISelectController])
@@ -14,4 +14,5 @@
     .controller("UITextController", ["$scope", "$attrs", "progressConfig", UITextController])
     .controller("UITextAreaController", ["$scope", "$attrs", "progressConfig", UITextAreaController])
     .controller("UIChartItemController", ["$scope", "$attrs", "progressConfig", "$timeout", UIChartItemController])
-    .controller("UIChartOptionController", ["$scope", "$attrs", "progressConfig", "$timeout", UIChartOptionController]);
+    .controller("UIChartOptionController", ["$scope", "$attrs", "progressConfig", "$timeout", UIChartOptionController])
+    .controller("UIAnchoredModalController", ["$scope", "$attrs", "progressConfig", "$timeout", UIAnchoredModalController]);
diff --git a/ngapp/app/controller/ApplicationController.js b/ngapp/app/controller/ApplicationController.js
index d2a387d..fd46d8e 100644
--- a/ngapp/app/controller/ApplicationController.js
+++ b/ngapp/app/controller/ApplicationController.js
@@ -16,6 +16,7 @@
             userName: 'Yogesh Surendran',
             greetingKey: 'Mr. ',
             troubleList: [],
+            searchEntry: '',
             loadTroubleList: function(){
                 $http
                 .get('http://' + location.hostname + ':8080/getScenarioList')
diff --git a/ngapp/app/controller/ConfigController.js b/ngapp/app/controller/ConfigController.js
index c67fef2..8068019 100644
--- a/ngapp/app/controller/ConfigController.js
+++ b/ngapp/app/controller/ConfigController.js
@@ -1,9 +1,24 @@
-function ConfigController($scope, $route, $http, $localStorage) {
+function ConfigController($scope, $route, $http, $localStorage, $routeParams) {
     $scope.portalValueField = 'portalNm';
     $scope.environmentValueField = 'environmentNm';
     $scope.projectValueField = 'moduleNm';
     
     $scope.$watch('editor.portal', function(newValue, oldValue){        
+        var idValue = -1;
+        angular.forEach($scope.application.portalList, function(value, key){
+            console.log(arguments, "arguments 2");
+            if(value.portalNm === newValue){
+                idValue = value.portalId;
+            }
+        });
+        if(idValue !== -1){
+            $scope.loadEnvironmentList(idValue);
+            $scope.loadProjectList(idValue);        
+            console.log(arguments, "arguments");
+        }
+    });
+    
+    $scope.$watch('selectedPortal', function(newValue, oldValue){        
         var idValue = -1;
         angular.forEach($scope.application.portalList, function(value, key){
             console.log(arguments, "arguments 2");
@@ -27,7 +42,7 @@
                     if(data.length > 0){
                         $scope.loadEnvironmentList(data[0].portalId);
                         $scope.loadProjectList(data[0].portalId); 
-                        console.log($scope.application.portalList);
+//                        console.log($scope.application.portalList);
                     }
                 })
                 .error(function (data, status, headers, config) {
@@ -41,7 +56,7 @@
                 .get('http://' + location.hostname + ':8080/getEnvironmentList?portal_id=' + param)
                 .success(function (data, status, headers, config) {
                     $scope.application.environmentList = data.data.result;
-                    console.log($scope.application.environmentList, data);
+//                    console.log($scope.application.environmentList, data);
                 })
                 .error(function (data, status, headers, config) {
                     $scope.application.model.toggle("We are unable to connect to the playbook webservice. Please try after some time.");
@@ -54,13 +69,24 @@
                 .get('http://' + location.hostname + ':8080/getModuleList?portal_id=' + param)
                 .success(function (data, status, headers, config) {
                     $scope.application.projectList = data.data.result;
-                    console.log($scope.application.projectList);
+//                    console.log($scope.application.projectList);
                 })
                 .error(function (data, status, headers, config) {
                     $scope.application.model.toggle("We are unable to connect to the playbook webservice. Please try after some time.");
                 });
         }
     };
+    $scope.handleSubmit = function(param){        
+        $http
+            .get('http://' + location.hostname + ':8080/searchScenario?query=' + $scope.data.searchEntry)
+            .success(function (data, status, headers, config) {
+                $scope.data.troubleList = data.data.result;
+//                    console.log($scope.application.projectList);
+            })
+            .error(function (data, status, headers, config) {
+                $scope.application.model.toggle("We are unable to connect to the playbook webservice. Please try after some time.");
+            });        
+    };
     
     $scope.loadPortalList();
 }
\ No newline at end of file
diff --git a/ngapp/app/controller/SceneController.js b/ngapp/app/controller/SceneController.js
index 9c6f8e4..dca5412 100644
--- a/ngapp/app/controller/SceneController.js
+++ b/ngapp/app/controller/SceneController.js
@@ -1,5 +1,6 @@
 function SceneController($scope, $route, $http, $localStorage){
-    $scope.selectCard = function(item, pList, editor){
+    $scope.selectCard = function(event, item, pList, editor){
+        console.log(event, arguments);
         angular.forEach(pList, function(yList, yKey){
             if(angular.isArray(yList)){
                 angular.forEach(yList, function(inst, key){
@@ -17,6 +18,8 @@
         if(angular.isDefined(editor)){
             editor.internal.selected = item;
             editor.editList = [item];
+            
+            editor.item.toggle(event.target);
         }
         
         console.log($scope, $scope.control);
diff --git a/ngapp/app/controller/UIAnchoredModalController.js b/ngapp/app/controller/UIAnchoredModalController.js
new file mode 100644
index 0000000..d452f33
--- /dev/null
+++ b/ngapp/app/controller/UIAnchoredModalController.js
@@ -0,0 +1,25 @@
+function UIAnchoredModalController($scope, $attrs, progressConfig, $timeout) {
+    $scope.modal = {};
+    
+//    console.log($scope.overlayclick);
+//    $scope.$watch($attrs.show, function(){
+//        console.log($attrs, $attrs.show);
+//        $scope.modal.show = $attrs.show;
+//    });
+//    $scope.$watch($attrs.style, function(){
+//        console.log($attrs.style);
+//        $scope.modal.style = $attrs.style;
+//    });
+//    $scope.$watch($attrs.anchor, function(){
+//        console.log($attrs.anchor);
+//        $scope.modal.anchor = $attrs.anchor;
+//    });
+//    $scope.$watch($attrs.title, function(){
+//        console.log($attrs.title);
+//        $scope.modal.title = $attrs.title;
+//    });
+//    $scope.$watch($attrs.overlayclick, function(){
+//        console.log($attrs.overlayclick);
+//        $scope.modal.overlayclick = $attrs.overlayclick;
+//    });
+}
\ No newline at end of file
diff --git a/ngapp/app/controller/UISelectController.js b/ngapp/app/controller/UISelectController.js
index 7f1c70e..7c76c24 100644
--- a/ngapp/app/controller/UISelectController.js
+++ b/ngapp/app/controller/UISelectController.js
@@ -18,7 +18,7 @@
     }
     $scope.expandList = function(){
         $scope.expand = !$scope.expand; 
-//        console.log("$scope.valueField", $attrs, angular.isDefined($scope.valuefield), $scope.valuefield);
+        console.log("$scope.valueField", $attrs, angular.isDefined($scope.valuefield), $scope.valuefield);
     }
     $scope.isEnabled = function(){
         return angular.isDefined($scope.valuefield);
