diff --git a/ngapp/app/controller/ConfigController.js b/ngapp/app/controller/ConfigController.js
index e1ca49d..c67fef2 100644
--- a/ngapp/app/controller/ConfigController.js
+++ b/ngapp/app/controller/ConfigController.js
@@ -21,7 +21,7 @@
     $scope.loadPortalList = function(){
         if($scope.application.portalList.length === 0){
             $http
-                .get('http://localhost:8080/getPortalList')
+                .get('http://' + location.hostname + ':8080/getPortalList')
                 .success(function (data, status, headers, config) {
                     $scope.application.portalList = data;
                     if(data.length > 0){
@@ -38,7 +38,7 @@
     $scope.loadEnvironmentList = function(param){
         if($scope.application.environmentList.length === 0){
             $http
-                .get('http://localhost:8080/getEnvironmentList?portal_id=' + param)
+                .get('http://' + location.hostname + ':8080/getEnvironmentList?portal_id=' + param)
                 .success(function (data, status, headers, config) {
                     $scope.application.environmentList = data.data.result;
                     console.log($scope.application.environmentList, data);
@@ -51,7 +51,7 @@
     $scope.loadProjectList = function(param){
         if($scope.application.projectList.length === 0){
             $http
-                .get('http://localhost:8080/getModuleList?portal_id=' + param)
+                .get('http://' + location.hostname + ':8080/getModuleList?portal_id=' + param)
                 .success(function (data, status, headers, config) {
                     $scope.application.projectList = data.data.result;
                     console.log($scope.application.projectList);
diff --git a/ngapp/app/controller/HomeController.js b/ngapp/app/controller/HomeController.js
index e1ff086..735c933 100644
--- a/ngapp/app/controller/HomeController.js
+++ b/ngapp/app/controller/HomeController.js
@@ -6,7 +6,7 @@
     $scope.loadScenarioList = function(param){
         if($scope.chart.chartList.length === 0){
             $http
-                .get('http://localhost:8080/getScenarioList?scenarioId=' + param)
+                .get('http://' + location.hostname + ':8080/getScenarioList?scenarioId=' + param)
                 .success(function (data, status, headers, config) {
                     if(data.data.result.length > 0){                        
                         var record = data.data.result[0];
diff --git a/ngapp/app/controller/SceneController.js b/ngapp/app/controller/SceneController.js
index faee6e2..9c6f8e4 100644
--- a/ngapp/app/controller/SceneController.js
+++ b/ngapp/app/controller/SceneController.js
@@ -48,19 +48,21 @@
         (enabled) ? ct.removeClass('jfc-hide') : ct.addClass('jfc-hide');
     };
     
-    $scope.manageConfig = function(){
-        $scope.editor.internal.toggleConfigEditor();
+    $scope.manageConfig = function(){          
+        $scope.application.anchoredModal.toggle();
     };
     
     $scope.addOption = function(){
         var scope = this;
         $scope.editor.internal.selected = scope.item;
+        $scope.editor.internal.enabled = true;
         $scope.$parent.addOption();
     };
     
     $scope.addActor = function(){
         var scope = this;
         $scope.editor.internal.selected = scope.item;
+        $scope.editor.internal.enabled = true;
         $scope.$parent.addActor();
     };
 }
\ No newline at end of file
diff --git a/ngapp/app/services/ChartService.js b/ngapp/app/services/ChartService.js
new file mode 100644
index 0000000..e69de29
--- /dev/null
+++ b/ngapp/app/services/ChartService.js
