diff --git a/ngapp/app/controller/ApplicationController.js b/ngapp/app/controller/ApplicationController.js
index c41c64d..4597dcd 100644
--- a/ngapp/app/controller/ApplicationController.js
+++ b/ngapp/app/controller/ApplicationController.js
@@ -1,4 +1,4 @@
-function ApplicationController($scope, $http, $location,$timeOut, $rootScope,  $localStorage, $sessionStorage){
+function ApplicationController($scope, $http, $location,$timeOut, $rootScope,  $localStorage, $sessionStorage, progressConfig){
 //     $scope = this;
 	$scope.pageLoading = false;
 	/*
@@ -19,7 +19,7 @@
             searchEntry: '',
             loadTroubleList: function(){
                 $http
-                .get('http://' + location.hostname + ':8080/getScenarioList')
+                .get(progressConfig.urlPrefix + '/getScenarioList')
                 .success(function (data, status, headers, config) {
                     $scope.data.troubleList = data.data.result;
                     console.log($scope.data.troubleList);
@@ -29,7 +29,13 @@
                 });
             },
             loadTroubleChart: function(item){
-                $location.url("/Chart/" + item.scenarioId + "/" + item.urlNm);
+                console.log("loadTroubleChart", $rootScope.application.isAdmin);
+                var url = ($rootScope.application.isAdmin ? "/Admin/" : "/Chart/") + item.scenarioId + "/" + item.urlNm;
+                if(location.hash == ('#!' + url)){
+                    history.go(0);
+                    return;
+            }
+                $location.url(url);
             }
 	};
 
@@ -60,6 +66,25 @@
                 }
                 return id;
             },
+            getEnvironmentValue: function(id){
+                return $scope.application.getValue($scope.application.environmentList, id, "environmentId", "environmentNm")
+            },
+            getProjectValue: function(id){
+                return $scope.application.getValue($scope.application.projectList, id, "moduleId", "moduleNm")
+            },
+            getPortalValue: function(id){
+                return $scope.application.getValue($scope.application.portalList, id, "portalId", "portalNm")
+            },
+            getValue: function(list, id, idField, valueField){
+                var value = "";
+                for(var i=0; i < list.length; i++){
+                    var obj = list[i];
+                    if(obj[idField] === id){
+                        value = obj[valueField];
+                    }
+                }
+                return value;
+            },
 	    setProject: false,
 	    removeProject: false,
 	    getProject: false,
@@ -69,6 +94,10 @@
             errorMsg:"",
             futureMsg:"Thanks for your interest This option will be added as part of next release!",
             notificationsCount:"",
+            isAdmin: false,
+            setAdmin: function(value){
+                $rootScope.application.isAdmin = value;
+            },
             model:{
                     errMsg:"",
                     visible: false,
