diff --git a/ngapp/app/controller/AdminController.js b/ngapp/app/controller/AdminController.js
index 62fe4c2..293321f 100644
--- a/ngapp/app/controller/AdminController.js
+++ b/ngapp/app/controller/AdminController.js
@@ -1,3 +1,13 @@
+/**
+ * @todo Implement Search, Copy Option, Edit functionality for existing flow, Filter options, Fix cosmetic issues.
+ * 
+ * @param {type} $scope
+ * @param {type} $route
+ * @param {type} $http
+ * @param {type} $timeout
+ * @param {type} $location
+ * @returns {undefined}
+ */
 function AdminController($scope, $route, $http, $timeout, $location){
 	$scope.editor = {
             chartList: [],
@@ -9,6 +19,7 @@
             name: "Chart Title",
             owner: "Build Team",
             internal: {
+                enabled: true,
                 editconfig: false,
                 toggleConfigEditor: function(){
                     $scope.editor.internal.editconfig = !$scope.editor.internal.editconfig;
@@ -26,7 +37,7 @@
             },
             editList:[], // used for editing only one record at a time. In rare case we may need to edit multpile items.
             toggleEditor: function(){
-                
+                $scope.editor.internal.enabled = !$scope.editor.internal.enabled;
             }
         };
         
@@ -217,7 +228,7 @@
             
             console.log(angular.toJson(param));
             $http({
-                url: 'http://localhost:8080/saveScenario',
+                url: 'http://' + location.hostname + ':8080/saveScenario',
                 method: 'POST',
                 transformRequest: function(obj) {
                     var str = [];
@@ -239,7 +250,8 @@
                 if(record === -1){
                     $scope.application.model.toggle("Chart Name '" + $scope.editor.name + "' has been already taken. Please choose another title.");
                 }else {
-                    window.open("#!/Chart/" + record.scenarioId + "/" + record.urlNm, "_blank");
+//                    window.open("#!/Chart/" + record.scenarioId + "/" + record.urlNm, "_blank");
+                    $scope.data.loadTroubleChart(record);
                 }                
 //                 $location.path("/Chart/" + record.scenarioId + "/" + record.urlNm);                
             }).error(function (data, status, headers, config) {
